lxd/provider.go

```diff
@@ -766,3 +757,39 @@ func determineDaemonAddr(lxdRemote terraformLXDConfig) (string, error) {
 
 	return daemonAddr, nil
 }
+
+// determineLxdDir determines which standard LXD directory contains a writable UNIX socket.
+// If environment variable LXD_DIR or LXD_SOCKET is set, the function will return LXD directory
+// based on the value from any of those variables.
+func determineLxdDir() (string, error) {
+	lxdSocket, ok := os.LookupEnv("LXD_SOCKET")
+	if ok {
+		return path.Dir(lxdSocket), nil
+	}
+
+	lxdDir, ok := os.LookupEnv("LXD_DIR")
+	if ok {
+		return lxdDir, nil
+	}
```

Comment:
Should we also check this directory for a writable socket, or just assume the user knows what they're doing?

Reply:
Good point, we could check both LXD_SOCKET and LXD_DIR (if any of those is set) to fail faster. I'll fix that.